### PR TITLE
job posting

### PR DESCRIPTION
**Pull request by Itsmeeeet**
- create job posting frontend based on figma.
- backend controller and route for jobPosting
- backend test creates a sample job, posts it to the database:
- Screenshot 2023-02-17 at 4 38 46 PM
-added frontend test:
Screenshot 2023-02-17 at 4 54 54 PM

### CODE REVIEW

File: src/server/package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "express": "^4.18.2",
     "express-async-handler": "^1.2.0",
     "express-session": "^1.17.3",
-    "mongoose": "^6.8.3"
+    "mongoose": "^6.8.3",
+    "multer": "^1.4.5-lts.1"
```
Comment (Aboudik1452) [inline src/server/package.json:22]: Please install the `react-icons` package by running `npm install react-icons`, and make sure that it's on this file as a dependency. I faced an error without having it installed.

File: src/client/src/components/JobPosting.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      description: document.getElementById("job_description").value,
+      salary: document.getElementById("salary").value,
+      category: document.getElementById("category").value,
+      location: document.getElementById("location").value,
+    };
+    const response = await axios.post(
+      `http://localhost:9000/jobs/create`,
+      jobData
+    );
+    console.log(response.data);
```
Comment (Aboudik1452) [inline src/client/src/components/JobPosting.jsx:83]: Consider alerting the user that the job posting is successfully created.

File: src/client/src/css/JobPosting.css [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+.title {
+  align-items: center;
+  display: flex;
+  justify-content: space-around;
+  margin: 0px auto;
+  margin-left: 0%;
+  margin-right: -18%;
+}
```
Comment (Aboudik1452) [inline src/client/src/css/JobPosting.css:19]: We should add a margin for all input boxes.
I suggest the following: `margin: 10px auto`

**Review by Aboudik1452 — CHANGES_REQUESTED** [review]
Good work @Itsmeeeet!
Please review the comments I posted.

File: src/client/src/components/JobPosting.jsx (hunk 2 above)
Comment (imran1289-ah) [reply src/client/src/components/JobPosting.jsx:83]: is it also possible to move the createJob function and added between JobPosting and the return statement. It's a bit weird having it in the bottom

**Review by imran1289-ah — APPROVED** [review]
Good changes ! I left a comment as a suggestion to make in your code.

File: src/client/src/components/JobPosting.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (imran1289-ah) [inline src/client/src/components/JobPosting.jsx:83]: > is it also possible to move the createJob function and added between JobPosting and the return statement. It's a bit weird having it in the bottom
this

File: src/client/src/css/JobPosting.css (hunk 3 above)
Comment (Itsmeeeet) [reply src/client/src/css/JobPosting.css:19]: I agree but to keep the UI good I am keeping `margin: 5px auto`

File: src/client/src/components/JobPosting.jsx (hunk 4 above)
Comment (Itsmeeeet) [reply src/client/src/components/JobPosting.jsx:83]: For now, I don't see any option to move this. As it is for the frontend. If you have any option please suggest me

File: src/client/src/components/JobPosting.jsx (hunk 2 above)
Comment (Itsmeeeet) [reply src/client/src/components/JobPosting.jsx:83]: Done!

File: src/server/package.json (hunk 1 above)
Comment (Itsmeeeet) [reply src/server/package.json]: Done!

**Review by Aboudik1452 — APPROVED** [review]

**Pull request merged**